jcommune-service/src/main/java/org/jtalks/jcommune/service/transactional/TransactionalPostService.java

```diff
@@ -149,11 +150,14 @@ public void deletePost(Post post) {
             branch.clearLastPost();
         }
 
-
         // todo: event API?
         topicDao.saveOrUpdate(topic);
         securityService.deleteFromAcl(post);
-        notificationService.subscribedEntityChanged(topic);
+
+        Set<JCUser> topicSubscribers = new HashSet<>(topic.getSubscribers());
+        topicSubscribers.remove(user);
+        notificationService.subscribedEntityChanged(topic, topicSubscribers);
```

Comment:
Maybe  we need to add a overloaded method which accept only a user? Foreach for subscribers is a bit slowly than work with one user.